### PR TITLE
Remove redemption restrictions when BorrowOperations contract has been removed from mintlist

### PR DESCRIPTION
**Pull request by benlongstaff**
Once the borrow operations contract for a type of collateral / version of contracts is removed from the thUSD token mintlist we need to relax the conditions around adjusting troves in that set of contracts

e.g. we have v1 contracts for TBTC and there are two troves

BTC = $120,000
_______________
trove 1 
collateral - 1 TBTC
debt - $100,000
ICR - 120%
_______________
trove 2
collateral - 1 TBTC
debt - $60,000
ICR - 200%
_______________
TCR - 150%

in this scenario trove 1 can have collateral withdrawn by the owner as it will raise the TCR however trove 2 cannot as the TCR would fall below the CCR (150%) 

After the BorrowerOperations contract is removed from the thUSD mintlist, the checks in 

_requireValidAdjustmentInCurrentMode.         (_adjustTrove)

and 

_requireNewTCRisAboveCCR              (closeTrove)
_requireNotInRecoveryMode.              (closeTrove)

need to be ignored

It's worth noting that because the borrower operations is removed from the mintlist that the _adjustTrove (used by adjustTrove, addColl, moveCollateralGainToTrove, withdrawColl, withdrawTHUSD, repayTHUSD) will revert if a borrower operation attempts to mint more

Some terminology used in the tests.

ICR - individual collateral ratio
TCR - total collateral ratio
MCR - minimum collateral ratio (110%)
CCR - Critical Collateralization Ratio (150%)
Recovery Mode = TCR < CCR

### CODE REVIEW

**Review by evandrosaturnino — APPROVED** [review]
LGTM

File: packages/contracts/contracts/BorrowerOperations.sol [hunk 1]
```diff
@@ -332,17 +333,20 @@ contract BorrowerOperations is LiquityBase, Ownable, CheckContract, IBorrowerOpe
...
         troveManagerCached.applyPendingRewards(msg.sender);
 
         uint256 coll = troveManagerCached.getTroveColl(msg.sender);
         uint256 debt = troveManagerCached.getTroveDebt(msg.sender);
 
         _requireSufficientTHUSDBalance(thusdTokenCached, msg.sender, debt - THUSD_GAS_COMPENSATION);
-
-        uint256 newTCR = _getNewTCRFromTroveChange(coll, false, debt, false, price);
-        _requireNewTCRisAboveCCR(newTCR);
+        if (thusdTokenCached.mintList(address(this))) {
```
Comment (vzotova) [inline packages/contracts/contracts/BorrowerOperations.sol]: this line and line 336 will lead to 2 intercontract calls which is not cheap, but value can't be changed between them. How about to cache value from first call?

File: packages/contracts/contracts/BorrowerOperations.sol [hunk 2]
```diff
@@ -522,19 +526,28 @@ contract BorrowerOperations is LiquityBase, Ownable, CheckContract, IBorrowerOpe
         internal
         view
     {
+        ITHUSDToken thusdTokenCached = thusdToken;
```
Comment (vzotova) [inline packages/contracts/contracts/BorrowerOperations.sol]: could you elaborate what is this cached value for?

File: packages/contracts/contracts/TroveManager.sol [hunk 3]
```diff
@@ -1211,10 +1208,13 @@ contract TroveManager is LiquityBase, Ownable, CheckContract, ITroveManager {
     }
 
     function _closeTrove(address _borrower, Status closedStatus) internal {
+        ITHUSDToken thusdTokenCached = thusdToken;
```
Comment (vzotova) [inline packages/contracts/contracts/TroveManager.sol]: One more place with unused caching

**Review by vzotova — APPROVED** [review]

**Pull request merged**